src/clj/game/core/rules.clj

```diff
@@ -423,21 +423,44 @@
                                       (swap! state update-in [:trash :trash-prevent] dissoc ktype)
                                       (effect-completed state side eid))
                                   (do (system-msg state :runner (str "will not prevent the trashing of " (:title card)))
-                                      (apply resolve-trash state side eid card args targets))))
+                                      (swap! state update-in [:trash :trash-list] concat [card])
+                                      (effect-completed state side eid))))
                               {:priority 10}))
-             ;; No prevention effects; resolve the trash.
-             (apply resolve-trash state side eid card args targets)))))
+             ;; No prevention effects: add the card to the trash-list
+             (do (swap! state update-in [:trash :trash-list] concat [card])
+                 (effect-completed state side eid))))))
      (effect-completed state side eid))))
 
+(defn trash
+  "Attempts to trash the given card, allowing for boosting/prevention effects."
+  ([state side card] (trash state side (make-eid state) card nil))
+  ([state side card args] (trash state side (make-eid state) card args))
+  ([state side eid {:keys [zone type] :as card} {:keys [unpreventable cause suppress-event] :as args}]
+   (swap! state update-in [:trash] dissoc :trash-list)
+   (when-completed (prevent-trash state side card args)
+                   (let [c (first (get-in @state [:trash :trash-list]))]
+                     (if c
```

Comment:
This should probably be an `if-let`  so `(if-let [c (first (get-in @state [:trash :trash-list]))] `etc